staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -747,8 +748,11 @@ func (v *specStandardValidatorV3) validate(schema *apiextensions.JSONSchemaProps
 		if v.allowDefaults {
 			if s, err := structuralschema.NewStructural(schema); err == nil {
 				// ignore errors here locally. They will show up for the root of the schema.
-				pruned := runtime.DeepCopyJSONValue(*schema.Default)
-				pruning.Prune(pruned, s)
+				pruned := runtime.DeepCopyJSONValue(interface{}(*schema.Default))
+				pruning.Prune(pruned, s, false)
+				if err := schemaobjectmeta.Coerce(fldPath, pruned, s, false, false); err != nil {
+					allErrs = append(allErrs, err)
+				}
 				if !reflect.DeepEqual(pruned, *schema.Default) {
 					allErrs = append(allErrs, field.Invalid(fldPath.Child("default"), schema.Default, "must not have unspecified fields"))
 				}
```

Comment:
It can't. See my comment in Slack. I have changed this to forbid defaults inside of metadata. We could relax this, this would require to construct an ObjectMeta object here and check that. Note that default can be arbitrarily deep in the ObjectMeta structure.

Alternatively, we could allow defaults in metadata, but be relaxed about validating them (other than for other defaults).

Reply:
This is postponed to https://github.com/kubernetes/kubernetes/pull/78788 for 1.16.